e2e-tests/demand-backup-incremental/run

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+
+set -o errexit
+
+test_dir=$(realpath "$(dirname "$0")")
+. "${test_dir}/../functions"
+set_debug
+
+run_backup() {
+	local storage=$1
+	local backup_name=$2
+	local base=${3:-true}
+
+	local backup_type="incremental"
+
+	if [[ $base == "true" ]]; then
+		backup_type="incremental-base"
+	fi
+
+	yq "$test_dir/conf/backup.yml" \
+		| $sed -e "s/name:/name: ${backup_name}/" \
+		| $sed -e "s/storageName:/storageName: ${storage}/" \
+		| yq ".spec.type=${backup_type}" \
+		| kubectl_bin apply -f -
+}
+
+run_restore() {
+	local backup_name=$1
+
+	echo "drop collection"
+	run_mongo 'use myApp\n db.test.drop()' "myApp:myPass@${cluster}-rs0.${namespace}"
+	echo "check backup and restore -- $backup_name"
+
+	yq "$test_dir/conf/restore.yml" \
+		| $sed -e "s/name:/name: restore-${backup_name}/" \
+		| $sed -e "s/backupName:/backupName: ${backup_name}/" \
+		| kubectl_bin apply -f -
+}
+
+run_recovery_check() {
+	local backup_name=$1
+	local compare_suffix=${2:-"_restore"}
+	local base=${3:-true}
+
+	wait_restore "${backup_name}" "${cluster}" "requested" "0" "3000"
+	echo
+
+	compare_kubectl "statefulset/${cluster}-rs0" "${compare_suffix}"
+
+	# we don't wait for cluster readiness here because the annotation gets removed then
+	wait_restore "${backup_name}" "${cluster}" "ready" "0" "1800"
+
+	if [ $(kubectl_bin get psmdb ${cluster} -o yaml | yq '.metadata.annotations."percona.com/resync-pbm"') == null ]; then
+		echo "psmdb/${cluster} should be annotated with percona.com/resync-pbm after a incremental restore"
+		exit 1
+	fi
+	echo
+
+	wait_cluster_consistency ${cluster}
+	wait_for_pbm_operations ${cluster}
+
+	if [[ $base == true ]]; then
+		compare_mongo_cmd "find" "myApp:myPass@${cluster}-rs0-0.${cluster}-rs0.${namespace}"
+		compare_mongo_cmd "find" "myApp:myPass@${cluster}-rs0-1.${cluster}-rs0.${namespace}"
+		compare_mongo_cmd "find" "myApp:myPass@${cluster}-rs0-2.${cluster}-rs0.${namespace}"
+	else
+		compare_mongo_cmd "find" "myApp:myPass@${cluster}-rs0-0.${cluster}-rs0.${namespace}" "-not-base"
+		compare_mongo_cmd "find" "myApp:myPass@${cluster}-rs0-1.${cluster}-rs0.${namespace}" "-not-base"
+		compare_mongo_cmd "find" "myApp:myPass@${cluster}-rs0-2.${cluster}-rs0.${namespace}" "-not-base"
+	fi
+}
+
+create_infra "${namespace}"
+
+deploy_minio
+apply_s3_storage_secrets
+
+desc 'Testing on not sharded cluster'
+
+echo "Creating PSMDB cluster"
```

Comment:
please use `log` function instead of `echo`

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1836/commits/f53376062c271385d7b4d9177498ddf727ad335e